chapters/operatorsandexpressions.tex

```diff
@@ -531,12 +535,26 @@ \subsection{Numeric Functions and Conversion Functions}\label{numeric-functions-
   \lstinline!(if leftJustified then "-" else "") + String(minimumLength) + "d"!
 \end{itemize}
 
-Form of the \lstinline!format! string:
-According to ANSI-C the format string specifies one conversion specifier (excluding the leading \%), shall not contain length modifiers, and shall not use `\lstinline!*!' for width and/or precision.
-For all numeric values the format specifiers `\lstinline!f!', `\lstinline!e!', `\lstinline!E!', `\lstinline!g!', `\lstinline!G!' are allowed.
-For integral values it is also allowed to use the `\lstinline!d!', `\lstinline!i!', `\lstinline!o!', `\lstinline!x!', `\lstinline!X!', `\lstinline!u!', and `\lstinline!c!' format specifiers (for non-integral values a tool may round, truncate or use a different format if the integer conversion characters are used).
+The ANSI-C style \lstinline!format! string (which cannot be combined with any of the other named arguments) consists of a single conversion specification without the leading \%.
+It shall not contain a length modifier, and shall not use `\lstinline!*!' for width and/or precision.
+For both \lstinline!Real! and \lstinline!Integer! values, the conversion specifiers `\lstinline!f!', `\lstinline!e!', `\lstinline!E!', `\lstinline!g!', `\lstinline!G!' are allowed.
+For \lstinline!Integer! values it is also allowed to use the `\lstinline!d!', `\lstinline!i!', `\lstinline!o!', `\lstinline!x!', `\lstinline!X!', `\lstinline!u!', and `\lstinline!c!' conversion specifiers (for \lstinline!Real! values a tool may round, truncate or use a different format if the integer conversion specifiers are used).
```

Comment:
While many things have now been clarified, this remains problematic. We're not saying that `String(100 * Constants.pi, format = "X")` is allowed, but still say it is OK return pretty much anything in this case.  How can this be a meaningful specification?

When we're at it, cleaning up `String`, why not get rid of this with a deprecation?
```suggestion
For \lstinline!Integer! values it is also allowed to use the `\lstinline!d!', `\lstinline!i!', `\lstinline!o!', `\lstinline!x!', `\lstinline!X!', `\lstinline!u!', and `\lstinline!c!' conversion specifiers.
Using the \lstinling!Integer! conversion specifiers for a \lstinline!Real! value is a deprecated feature without defined result.
```

Reply:
I can see some benefit in doing that, but I think it is best to discuss it in a separate PR focused on that topic so that we can close this one.